cypress/support/dashboard_commands.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Add saved Dashboard panels
+ * @param {String} keyword Search term for panels of interest
+ * @param {String} type Panel type to search for
+ * @param {Boolean} multiplePages Whether there are multiple pages to iterate through
+ */
+
+Cypress.Commands.add('addDashboardPanels', (keyword, type, multiplePages = true) => {
+  const iteratePages = () => {
+    cy.get('[data-test-subj="pagination-button-next"]').then(($nextBtn) => {
+      cy.get('[data-test-subj="savedObjectFinderItemList"]').should('be.visible')
+      if ($nextBtn.is(':enabled')) {
+        cy.wrap($nextBtn).click()
+        cy.get('[data-test-subj^="savedObjectTitle' + keyword.replace(/\s+/g, '-') + '"]').should('be.visible').each(($button) => {
+          cy.wrap($button).click()
+          cy.get('[data-test-subj="toastCloseButton"').click({ multiple: true, force: true })
+        })
+        iteratePages()
+      }
+    })
+  }
+  cy.get('[data-test-subj="dashboardAddPanelButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderItemList"]').should('be.visible')
+
+  cy.get('[data-test-subj="savedObjectFinderFilterButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderFilter-' + type + '"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderFilterButton"]').should('be.visible').click()
+
+  cy.get('[data-test-subj="savedObjectFinderSearchInput"]').should('be.visible').type(keyword)
+
+  cy.get('[data-test-subj^="savedObjectTitle' + keyword.replace(/\s+/g, '-') + '"]').should('be.visible').each(($button) => {
```

Comment:
refactory below piece code to a method for readability

``keyword.replace(/\s+/g, '-') + '"]'``

Reply:
I now refactored the specified code to its own constant function (`replaceSpacesWithDashes`).